src/shared/fstab-util.c

```diff
@@ -34,6 +34,21 @@
 #include "strv.h"
 #include "util.h"
 
+bool fstab_has_fstype(const char *fstype) {
+        _cleanup_endmntent_ FILE *f = NULL;
+        struct mntent *m;
+
+        f = setmntent("/etc/fstab", "re");
+        if (!f)
```

Comment:
following the logic above, I think the error handling should look something like this:

```c
if (!f) 
        return errno == ENOENT ? false : -errno;
```

(as a missing /etc/fstab should be considered equivalent to an empty one, but any other error should be propagated)